Sources/Utility/ArgumentParser.swift

```diff
@@ -459,17 +558,48 @@ public final class ArgumentParser {
         usage: String? = nil
     ) -> PositionalArgument<T> {
         precondition(subparsers.isEmpty, "Positional arguments are not supported with subparsers")
-        precondition(optional ? positionalArgs.count <= 1 : true,
-            "Only one positional argument is allowed if optional setting is enabled.")
-        optionalPositionalArg = optional
-        let arg = PositionalArgument<T>(name: positional, usage: usage)
-        positionalArgs.append(AnyArgument(arg))
-        return arg
+        precondition(canAcceptPositionalArguments, "Can not accept more positional arguments")
+
+        if optional {
+            canAcceptPositionalArguments = false
+        }
+
+        let argument = PositionalArgument<T>(name: positional, usage: usage)
+        positionalArguments.append(ParserPositionalArgument(
+            argument: AnyArgument(argument),
+            strategy: .oneByOne,
+            isOptional: optional))
+        return argument
+    }
+
+    /// Adds an argument to the parser.
+    /// Note: Only one multiple-value positional argument is allowed.
+    public func add<T: ArgumentKind>(
+        positional: String,
+        kind: [T].Type,
+        optional: Bool = false,
+        strategy: ArrayParsingStrategy = .upToNextOption,
+        usage: String? = nil
+    ) -> PositionalArgument<[T]> {
+        precondition(subparsers.isEmpty, "Positional arguments are not supported with subparsers")
+        precondition(canAcceptPositionalArguments, "Can not accept more positional arguments")
+
+        if optional || strategy == .remaining {
+            canAcceptPositionalArguments = false
+        }
+
+        let argument = PositionalArgument<[T]>(name: positional, usage: usage)
+        positionalArguments.append(ParserPositionalArgument(
+            argument: AnyArgument(argument),
+            strategy: strategy,
+            isOptional: optional))
```

Comment:
We should have an assert in init of `ParserPositionalArgument` that argument is an array if strategy is not `.oneByOne`